### PR TITLE
Add optional title

### PR DESCRIPTION
**Pull request by joannaquu**
### Platforms Impacted
- [x] iOS
- [ ] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Added support for an optional title. If no title is provided, the message body uses the title's style.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| 6 - Button:Pressable | 7 - Title |

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Notification/src/Notification.styling.ts [hunk 1]
```diff
@@ -49,6 +58,22 @@ export const stylingSettings: UseStylingOptions<NotificationProps, NotificationS
...
+            fontWeight: tokens.fontWeight,
+            lineHeight: tokens.fontLineHeight,
+            letterSpacing: tokens.fontLetterSpacing,
+            flex: 1,
+            flexGrow: 1,
+          },
+        };
+      },
+      ['color'],
+    ) as TextStyle,
```
Comment (joannaquu) [inline packages/components/Notification/src/Notification.styling.ts:76]: Is this an ok way to do it? I added this to address the { ... } is not assignable to type 'TextStyle' error

Comment (Saadnajmi) [reply packages/components/Notification/src/Notification.styling.ts:76]: What's the full error?

Comment (joannaquu) [reply packages/components/Notification/src/Notification.styling.ts:76]: The types of 'style' are incompatible between these types.

Type '{ color: ColorValue; fontSize: number | keyof FontSizes; fontWeight: "bold" | "600" | keyof FontWeights | "normal" | "100" | "200" | "300" | ... 4 more ... | "900"; lineHeight: number; letterSpacing: number; flex: number; flexGrow: number; }' is not assignable to type 'StyleProp<TextStyle>'.

Type '{ color: ColorValue; fontSize: number | keyof FontSizes; fontWeight: "bold" | "600" | keyof FontWeights | "normal" | "100" | "200" | "300" | ... 4 more ... | "900"; lineHeight: number; letterSpacing: number; flex: number; flexGrow: number; }' is not assignable to type 'TextStyle'.

Comment (Saadnajmi) [reply packages/components/Notification/src/Notification.styling.ts:76]: Talked and solved offline. Solution is to use `...fontStyles.from(tokens, theme)`

**Review by Saadnajmi — APPROVED** [review]

**Pull request merged**